### ISSUE TITLE
Prevent unexcepted usage of SocialNetworkUIDefinition > posts

### ISSUE THREAD

**Issue by Misaka-0x447f** [DimensionDev/Maskbook#254]
In the current version (3bf6b3b3afd35d58c76c37f947048f3c69c17f44) of maskbook social-network definition, either facebook and twitter using the key of ```posts``` passing information(root node of this post) which should be not -- According to description of @Jack-Works , the key of post should be never used but in the typing system it is used without explicit definition.
The current type of post is:
```typescript
    readonly posts?: WeakMap<DOMProxy, PostInfo>
```
To fix this, we can set type of key to:
```typescript
    readonly posts?: WeakMap<[], PostInfo>
```

### how it used since iterate key of WeakMap is not possible?
Some code using it by rewrite the set and delete function.
https://github.com/DimensionDev/Maskbook/blob/3bf6b3b3afd35d58c76c37f947048f3c69c17f44/src/social-network/ui.ts#L258-L281

**Comment by Misaka-0x447f** [comment]
#228

**Comment by Jack-Works** [comment]
`readonly posts?: WeakMap<object, PostInfo>` will be better

**Comment by Misaka-0x447f** [comment]
Why not narrow it to an empty object...?
readonly posts?: WeakMap<{}, PostInfo>

**Comment by Jack-Works** [comment]
closed by #256
